articles/Intro to C++ - Part 3/index.md

```diff
@@ -0,0 +1,141 @@
+
+## Functions in C++
+
+### **What are Functions?**
+
+Functions have a name and take zero or more parameters. The parameters are how information gets sent to the function. The function only cares about the order of the parameters. It doesn't know or care about the names of the variables in your program. 
+
+The only way to send information to a function is through parameters. And the only way a function can send information back is by returning the value.
+
+**Why do we need function?**
+
+* To write a task once that you'll use a bunch of times.
+* To break the task of writing a big program into smaller, more easily handleable pieces.
+* To create a library of operations for use in multiple programs.
+
+Let's write a function that returns the largest of the three parameters:
+
+```C
+int max_of_three (int one, int two, int three) {
+
+//If one is greater than or equal to two and three: return one
+if (one>=two && one>=three)
+   return one;
+
+//If two is greater than or equal to one and three: return two
+else if (two>=one && two>=three)
+  return two;
+
+//If none of the above statements is true, then three would be the highest: return three
+else
+  return three;
+}
+```
+The function max_of_three would take three parameters and return the largest of the three parameters.
+
+Now let's see how to call (implement) this function in main. 
+
+```C
+#include<iostream>
+using namespace std;
+int main() {
+cout<<"enter the three number to compare:"<<endl;
+int one, two, three;
+cin>>one>>two>>three;
+
+//Yes, Just that's how you can call any function in the main
+max_of_three (one, two, three);
+
+return 0;
+}
+```
+That's how we write and implement functions in C++. 
+
+Let's go through another example of functions where we will be printing a rectangle of * character.
+
+When writing functions, you should ask yourself some questions.
+
+1. What should be the name of the function?
+2. What parameter should I use?
+3. What do I return?
+
+```C
+void print_rectangle (int num_rows, int num_cols) {
+    for (int i=0; i<num_rows; i++){
+       for (int j=0; j<num_cols; j++){
+             cout<<"*";
+       }
+cout<<endl;
+    } 
+}
+```
+
+Now let's see how to call (implement) this function in main. 
+
+```C
+#include<iostream>
+using namespace std;
+int main() {
+
+cout<<"enter the number of rows:"<<endl;
+int rows;
+cin>>rows;
+
+cout<<"enter the number of columns:"<<endl;
+int cols;
+cin>>cols;
+
+//Now call the function
+print_rectangle (rows, cols);
+
+return 0;
+}
+```
+
+Let's see what the output would be, and then we will go into some details.
+
+```C
+Enter the number of rows: 5
+Enter the number of columns: 5
+
+*****
+*****
+*****
+*****
+*****
+```
+
+The only new thing for us in the above function is **void print_rectangle**. Usually, we write **int** but why **void** this time? 
+
+It's because of the fact that this function doesn't mean to return anything. And for these kinds of functions, we use **void** instead of **int**.
```

Comment:
It's because -> This is because

Reply:
doesn't mean to return anything -> doesn't return anything

Reply:
anything. And for these ->  anything. For these

Reply:
Add a line about how the return type matches the type of variable that gets returned